File: wfm-develop mobile 07.05.2026/wfm/mobile/ios/WFMUI/Sources/WFMUI/Components/WFMBadge.swift
import SwiftUI

// MARK: - Badge Color

/// Цвета бейджа
public enum BadgeColor {
    case violet
    case blue
    case yellow
    case pink
    case orange
    case green

    func backgroundColor(_ colors: WFMSemanticColors) -> Color {
        switch self {
        case .violet: return colors.badgeBrandBgLight
        case .blue: return colors.badgeBlueBgLight
        case .yellow: return colors.badgeYellowBgLight
        case .pink: return colors.badgePinkBgLight
        case .orange: return colors.badgeOrangeBgLight
        case .green: return colors.badgeGreenBgLight
        }
    }

    func textColor(_ colors: WFMSemanticColors) -> Color {
        switch self {
        case .violet: return colors.badgeBrandTextBright
        case .blue: return colors.badgeBlueTextBright
        case .yellow: return colors.badgeYellowTextBright
        case .pink: return colors.badgePinkTextBright
        case .orange: return colors.badgeOrangeTextBright
        case .green: return colors.badgeGreenTextBright
        }
    }
}

// MARK: - Badge Icon

/// Тип иконки в бейдже
public enum BadgeIcon {
    case none       // Без иконки
    case dot        // Точка слева
    case thunder    // Иконка молнии слева (для бонусов)
}

// MARK: - Badge Component

/// Компонент Badge дизайн-системы WFM
///
/// Используется для отображения статусов, категорий, бонусов и других меток.
///
/// - Parameters:
///   - text: Текст бейджа
///   - color: Цвет бейджа (определяет фон и цвет текста)
///   - icon: Тип иконки (none, dot, thunder)
public struct WFMBadge: View {
    let text: String
    let color: BadgeColor
    let icon: BadgeIcon

    @Environment(\.wfmColors) private var colors

    public init(text: String, color: BadgeColor, icon: BadgeIcon = .none) {
        self.text = text
        self.color = color
        self.icon = icon
    }

    public var body: some View {
        HStack(spacing: icon == .none ? 0 : WFMSpacing.xxxs) {
            // Иконка (если есть)
            switch icon {
            case .dot:
                Circle()
                    .fill(color.textColor(colors))
                    .frame(width: 4, height: 4)

            case .thunder:
                // TODO: Добавить иконку молнии (нужно экспортировать из Figma)
                // Временно используем заглушку
                Circle()
                    .fill(color.textColor(colors))
                    .frame(width: 16, height: 16)

            case .none:
                EmptyView()
            }

            // Текст
            Text(text)
                .font(WFMTypography.headline12Medium)
                .tracking(WFMTypography.LetterSpacing.headline12Medium)
                .foregroundColor(color.textColor(colors))
        }
        .padding(.horizontal, WFMSpacing.xs)
        .padding(.vertical, WFMSpacing.xxxs)
        .background(color.backgroundColor(colors))
        .cornerRadius(WFMRadius.s)
    }
}

// MARK: - Previews

#Preview("Badge - Violet (Выкладка)") {
    WFMBadge(text: "Выкладка", color: .violet)
        .padding()
        .wfmTheme()
}

#Preview("Badge - Blue (Работа на кассе)") {
    WFMBadge(text: "Работа на кассе", color: .blue)
        .padding()
        .wfmTheme()
}

#Preview("Badge - Yellow (Смена ценников)") {
    WFMBadge(text: "Смена ценников", color: .yellow)
        .padding()
        .wfmTheme()
}

#Preview("Badge - Pink (Другие работы)") {
    WFMBadge(text: "Другие работы", color: .pink)
        .padding()
        .wfmTheme()
}

#Preview("Badge - Orange (Другие работы)") {
    WFMBadge(text: "Другие работы", color: .orange)
        .padding()
        .wfmTheme()
}

#Preview("Badge - Green (Другие работы)") {
    WFMBadge(text: "Другие работы", color: .green)
        .padding()
        .wfmTheme()
}

#Preview("Badge - Orange + Thunder (Бонус)") {
    WFMBadge(text: "Бонус 300 ₽", color: .orange, icon: .thunder)
        .padding()
        .wfmTheme()
}

#Preview("Badge - Blue + Dot (Создано)") {
    WFMBadge(text: "Создано", color: .blue, icon: .dot)
        .padding()
        .wfmTheme()
}

#Preview("All Badges") {
    VStack(spacing: 12) {
        WFMBadge(text: "Выкладка", color: .violet)
        WFMBadge(text: "Работа на кассе", color: .blue)
        WFMBadge(text: "Смена ценников", color: .yellow)
        WFMBadge(text: "Другие работы", color: .pink)
        WFMBadge(text: "Другие работы", color: .orange)
        WFMBadge(text: "Другие работы", color: .green)
        WFMBadge(text: "Бонус 300 ₽", color: .orange, icon: .thunder)
        WFMBadge(text: "Создано", color: .blue, icon: .dot)
    }
    .padding()
    .wfmTheme()
}

File: wfm-develop mobile 07.05.2026/wfm/mobile/android/ui/src/main/kotlin/com/beyondviolet/wfm/ui/components/WfmBadge.kt
package com.beyondviolet.wfm.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import com.beyondviolet.wfm.ui.theme.WfmColors
import com.beyondviolet.wfm.ui.theme.WfmRadius
import com.beyondviolet.wfm.ui.theme.WfmSpacing
import com.beyondviolet.wfm.ui.theme.WfmTheme
import com.beyondviolet.wfm.ui.theme.WfmTypography
import com.beyondviolet.wfm.ui.theme.*

/**
 * Цвета бейджа
 */
enum class BadgeColor {
    VIOLET,
    BLUE,
    YELLOW,
    PINK,
    ORANGE,
    GREEN;

    fun backgroundColor(colors: com.beyondviolet.wfm.ui.theme.WfmSemanticColors): Color = when (this) {
        VIOLET -> colors.badgeBrandBgLight
        BLUE -> colors.badgeBlueBgLight
        YELLOW -> colors.badgeYellowBgLight
        PINK -> colors.badgePinkBgLight
        ORANGE -> colors.badgeOrangeBgLight
        GREEN -> colors.badgeGreenBgLight
    }

    fun textColor(colors: com.beyondviolet.wfm.ui.theme.WfmSemanticColors): Color = when (this) {
        VIOLET -> colors.badgeBrandTextBright
        BLUE -> colors.badgeBlueTextBright
        YELLOW -> colors.badgeYellowTextBright
        PINK -> colors.badgePinkTextBright
        ORANGE -> colors.badgeOrangeTextBright
        GREEN -> colors.badgeGreenTextBright
    }
}

/**
 * Тип иконки в бейдже
 */
enum class BadgeIcon {
    NONE,      // Без иконки
    DOT,       // Точка слева
    THUNDER    // Иконка молнии слева (для бонусов)
}

/**
 * Компонент Badge дизайн-системы WFM
 *
 * Используется для отображения статусов, категорий, бонусов и других меток.
 *
 * @param text Текст бейджа
 * @param color Цвет бейджа (определяет фон и цвет текста)
 * @param modifier Modifier
 * @param icon Тип иконки (NONE, DOT, THUNDER)
 */
@Composable
fun WfmBadge(
    text: String,
    color: BadgeColor,
    modifier: Modifier = Modifier,
    icon: BadgeIcon = BadgeIcon.NONE
) {
    val colors = WfmTheme.colors

    Row(
        modifier = modifier
            .background(
                color = color.backgroundColor(colors),
                shape = RoundedCornerShape(WfmRadius.S)
            )
            .padding(
                horizontal = WfmSpacing.XS,
                vertical = WfmSpacing.XXXS
            ),
        horizontalArrangement = Arrangement.spacedBy(
            if (icon == BadgeIcon.NONE) 0.dp else WfmSpacing.XXXS
        ),
        verticalAlignment = Alignment.CenterVertically
    ) {
        // Иконка (если есть)
        when (icon) {
            BadgeIcon.DOT -> {
                Box(
                    modifier = Modifier
                        .size(4.dp)
                        .background(
                            color = color.textColor(colors),
                            shape = CircleShape
                        )
                )
            }
            BadgeIcon.THUNDER -> {
                // TODO: Добавить иконку молнии (нужно экспортировать из Figma)
                // Временно используем заглушку
                Box(
                    modifier = Modifier
                        .size(16.dp)
                        .background(
                            color = color.textColor(colors),
                            shape = CircleShape
                        )
                )
            }
            BadgeIcon.NONE -> {
                // Без иконки
            }
        }

        // Текст
        Text(
            text = text,
            style = WfmTypography.Headline12Medium,
            color = color.textColor(colors),
            maxLines = 1,
            overflow = TextOverflow.Ellipsis
        )
    }
}

// ═══════════════════════════════════════════════════════════════════════════
// PREVIEWS
// ═══════════════════════════════════════════════════════════════════════════

@Preview(name = "Badge - Violet (Выкладка)", showBackground = true, backgroundColor = 0xFFF5F5FC)
@Composable
private fun WfmBadgeVioletPreview() {
    WfmTheme {
        WfmBadge(
            text = "Выкладка",
            color = BadgeColor.VIOLET
        )
    }
}

@Preview(name = "Badge - Blue (Работа на кассе)", showBackground = true, backgroundColor = 0xFFF5F5FC)
@Composable
private fun WfmBadgeCashierPreview() {
    WfmTheme {
        WfmBadge(
            text = "Работа на кассе",
            color = BadgeColor.BLUE
        )
    }
}

@Preview(name = "Badge - Yellow (Смена ценников)", showBackground = true, backgroundColor = 0xFFF5F5FC)
@Composable
private fun WfmBadgeYellowPreview() {
    WfmTheme {
        WfmBadge(
            text = "Смена ценников",
            color = BadgeColor.YELLOW
        )
    }
}

@Preview(name = "Badge - Pink (Другие работы)", showBackground = true, backgroundColor = 0xFFF5F5FC)
@Composable
private fun WfmBadgePinkPreview() {
    WfmTheme {
        WfmBadge(
            text = "Другие работы",
            color = BadgeColor.PINK
        )
    }
}

@Preview(name = "Badge - Orange (Другие работы)", showBackground = true, backgroundColor = 0xFFF5F5FC)
@Composable
private fun WfmBadgeOrangePreview() {
    WfmTheme {
        WfmBadge(
            text = "Другие работы",
            color = BadgeColor.ORANGE
        )
    }
}

@Preview(name = "Badge - Green (Другие работы)", showBackground = true, backgroundColor = 0xFFF5F5FC)
@Composable
private fun WfmBadgeGreenPreview() {
    WfmTheme {
        WfmBadge(
            text = "Другие работы",
            color = BadgeColor.GREEN
        )
    }
}

@Preview(name = "Badge - Orange + Thunder (Бонус)", showBackground = true, backgroundColor = 0xFFF5F5FC)
@Composable
private fun WfmBadgeBonusPreview() {
    WfmTheme {
        WfmBadge(
            text = "Бонус 300 ₽",
            color = BadgeColor.ORANGE,
            icon = BadgeIcon.THUNDER
        )
    }
}

@Preview(name = "Badge - Blue + Dot (Создано)", showBackground = true, backgroundColor = 0xFFF5F5FC)
@Composable
private fun WfmBadgeCreatedPreview() {
    WfmTheme {
        WfmBadge(
            text = "Создано",
            color = BadgeColor.BLUE,
            icon = BadgeIcon.DOT
        )
    }
}
